src/plugins/intel_npu/src/plugin/npuw/llm_infer_request.cpp

```diff
@@ -140,6 +143,17 @@ ov::npuw::LLMInferRequest::LLMInferRequest(const std::shared_ptr<ov::npuw::LLMCo
     }
 
     m_kvcache_request = compiled_model->m_kvcache_compiled->create_infer_request();
+
+    m_prefil_request_listener = std::make_shared<LLMPrefillInferRequest>(
+        [this](std::size_t idx, IInferRequestSubmissionListener::Completed cb){
+            this->subscribe_subrequest(idx, cb);
+        },
+        [this](ov::SoPtr<ov::IAsyncInferRequest> req, size_t idx){
+            this->complete_subrequest(req, idx);
+        }
```

Comment:
Subrequests shouldn't be exposed at this level.
This one should subscribe to the particular output(s) but not subrequests. Assume subrequests is a very private implementation detail at this level.

Once the output (e.g. 'k' or 'v') is complete, the callback is triggered. That's it

Reply:
yes the question is subscription mechanics - @dmatveev what would you suggest
1. We dont have access to real IbaseIreq in llm_infer_request, dynamic cast - not a good idea as well.
2. Add subscription interface into Compiled Model?  Prior creating infer-requests, since this is the only factory class for infer-requests we can configure - similar to add listerner i've introduced - we can add more detailed listener mechanics - listen by name pattern for example, or just all names, we interested and then, listen to each individually whenever it became ready?

Reply:
3. we can receive in subscriber all events for every output when it is ready, and then filter it - this minimizes subscription contract and can do more flexible filtering on llm_infer_request side i guess
4. ?